### DISCUSSION TITLE
why configuration otions are commented?

### DISCUSSION THREAD

**Discussion by harrykar** [ActivityWatch/activitywatch#1015] [Q&A] [answered]
After started the server after some hours I noticed there are no data collected. 
Opening all config files I notice all options are commented per default, why that?

So I have to uncomment all those lines right?

PS: I run a Linux box(MX Linux -Debian's derivative-). Downloaded 0.12.3b16 release
TIA

in aw-client.toml
```
[server]
#hostname = "127.0.0.1"
#port = "5600"

[client]
#commit_interval = 10
```

in aw-watcher-afk.toml
```
[aw-watcher-afk]
#timeout = 180
#poll_time = 5

```

in aw-watcher-window.toml
```
[aw-watcher-window]
#exclude_title = false
#poll_time = 1.0
#strategy_macos = "swift"
```

in aw-qt.toml
```
[aw-qt]
#autostart_modules = ["aw-server", "aw-watcher-afk", "aw-watcher-window"]
```

in aw-server.toml
```
[server]
#host = "localhost"
#port = "5600"
#storage = "peewee"
#cors_origins = ""
```

**Comment by BelKed** [comment, marked as answer]
In many cases, config files, including TOML files, come with default options commented out as a way to provide guidance to users about available options and their default values. It's a helpful practice because it makes it easier for users to understand what options are available and what their default values are without needing to consult extensive documentation. Plus, it encourages users to only uncomment and modify the options they need, reducing the chances of errors or unintentional changes.

So, there's no need to uncomment them, as long as you're not changing the values :)

I would suspect there's also an issue on Linux with the latest beta release (as on macOS).
I'd recommend you trying the latest stable version `0.12.2` — this should has minimum bugs since it's a stable one. Also, the previous beta version (`0.12.3b15`) should be working :)
(Of course, assuming that there isn't any other problem such as compatibility issues with your operating system.)

**Comment by harrykar** [comment]
![Screenshot_2024-01-28_23-24-26](https://github.com/ActivityWatch/activitywatch/assets/465842/cbca27f7-4917-459f-9479-85b710a22586)
![Screenshot_2024-01-28_23-24-05](https://github.com/ActivityWatch/activitywatch/assets/465842/1cd60e84-8f8b-4c95-b2dd-65fa8956538a)
@BelKed thanks for your ready reply 
In other words commented options represent defaults so is indifferent if remain commented but if I change something should uncomment right?
 
I just noticed browser's watcher outputs something but the rest is not responding
Furthermore tray icon is missing

Before I try a downgrade here is some output 

first(and actual) run with some errors
```
$ ./aw-qt
2024-01-28 19:28:44 [INFO ]: Started aw-qt...  (aw_qt.main:53)
2024-01-28 19:28:44 [WARNING]: Found matching file but was not executable: /home/harrykar/activitywatch/aw-server-rust/aw-server.service  (aw_qt.manager:64)
2024-01-28 19:28:44 [WARNING]: Found matching file but was not executable: /home/harrykar/activitywatch/aw-qt.desktop  (aw_qt.manager:64)
2024-01-28 19:28:44 [INFO ]: Found 7 bundled modules  (aw_qt.manager:85)
2024-01-28 19:28:44 [INFO ]: Found 0 system modules  (aw_qt.manager:119)
2024-01-28 19:28:44 [INFO ]: Starting module aw-server  (aw_qt.manager:148)
2024-01-28 19:28:44 [INFO ]: Starting module aw-watcher-window  (aw_qt.manager:148)
2024-01-28 19:28:44 [INFO ]: Starting module aw-watcher-afk  (aw_qt.manager:148)
2024-01-28 19:28:44 [INFO ]: Creating trayicon...  (aw_qt.trayicon:206)
qt.qpa.plugin: From 6.5.0, xcb-cursor0 or libxcb-cursor0 is needed to load the Qt xcb platform plugin.
qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "" even though it was found.
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: eglfs, linuxfb, vnc, wayland, minimalegl, offscreen, xcb, vkkhrdisplay, wayland-egl, minimal.

Aborted
harrykar@mx:~/activitywatch
$ 2024-01-28 19:28:44 [INFO ]: aw-watcher-afk started  (aw_watcher_afk.afk:62)
2024-01-28 19:28:44 [INFO ]: aw-watcher-window started  (aw_watcher_window.main:62)
2024-01-28 19:28:44 [INFO ]: Using storage method: peewee  (aw_server.main:33)
2024-01-28 19:28:44 [INFO ]: Starting up...  (aw_server.main:41)
2024-01-28 19:28:44 [INFO ]: Using database file: /home/harrykar/.local/share/activitywatch/aw-server/peewee-sqlite.v2.db  (aw_datastore.storages.peewee:150)
 * Serving Flask app 'aw-server'
 * Debug mode: off
Address already in use
Port 5600 is in use by another program. Either identify and stop that program, or start the server with a different port.
2024-01-28 19:28:45 [INFO ]: afkwatcher stopped because parent process died  (aw_watcher_afk.afk:82)
2024-01-28 19:28:45 [INFO ]: Connection to aw-server established by aw-watcher-afk  (aw_client.client:447)
2024-01-28 19:28:45 [INFO ]: window-watcher stopped because parent process died  (aw_watcher_window.main:101)
2024-01-28 19:28:45 [INFO ]: Connection to aw-server established by aw-watcher-window  (aw_client.client:447)

```


the server run flawlessly. How can stop, restart the server without killing it? 
```
$ ps -aux |grep aw-server
harrykar   51314  0.0  0.5 611608 45568 ?        S    19:23   0:02 /home/harrykar/activitywatch/aw-server/aw-server
harrykar   62016  0.0  0.0  75980  2176 pts/2    S+   20:30   0:00 grep aw-server

```

**Comment by BelKed** [comment]
> In other words commented options represent defaults so is indifferent if remain commented but if I change something should uncomment right?

Yup, exactly :)

---

From the logs I can see that the error causing problems with the tray icon is:

```
qt.qpa.plugin: From 6.5.0, xcb-cursor0 or libxcb-cursor0 is needed to load the Qt xcb platform plugin.
```

The following Stack Overflow answer seems to provide a solution to that error:
https://stackoverflow.com/questions/77725761

If you have any further questions, feel free to ask :)

**Comment by harrykar** [comment]
> > In other words commented options represent defaults so is indifferent if remain commented but if I change something should uncomment right?
> 
> Yup, exactly :)

Thank you I'm first-timer  in encountering a "pedagogical" configuration file :)

> From the logs I can see that the error causing problems with the tray icon is:
> 
> ```
> qt.qpa.plugin: From 6.5.0, xcb-cursor0 or libxcb-cursor0 is needed to load the Qt xcb platform plugin.
> ```
> 
> The following Stack Overflow answer seems to provide a solution to that error: https://stackoverflow.com/questions/77725761
> 
> If you have any further questions, feel free to ask :)

Thank you very much for your interest but that have to wait for a while because the other day after setting up the stable version of AW(I can confirm that work as intended out of the box) had a serious issue with my box and after restore it morning today  I try to devise what happened so wrong. Hoping that inconvenient be a 'random coincidence' and the  culprit not be AW. More I have reported [here] : https://forum.mxlinux.org/viewtopic.php?t=79116
